lisp/pdf-view.el

```diff
@@ -930,6 +932,9 @@ See also `pdf-view-use-imagemagick'."
                   (* 2 (car size)))))
          (hotspots (pdf-view-apply-hotspot-functions
                     window page size)))
+    (when (and (eq (framep-on-display) 'mac)
+               (= (pdf-util-frame-scale-factor) 2))
+      (put-text-property 0 1 :data-2x data data))
```

Comment:
I would prefer sticking it into the image as a property, if that is possible.

Reply:
OK I will do that. Also, I will move setting the property over into `pdf-view-create-image` since functions like `pdf-view-display-region` render a page but don't actually go through `pdf-view-create-page`.

Reply:
Thanks.